packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -133,14 +153,24 @@ export class BlockToolAdapter {
       case InputType.DeleteSoftLineForward:
       case InputType.DeleteWordBackward:
       case InputType.DeleteWordForward: {
+        if (nativeInput && start > 0 && start === end) {
+          // for native input elements, we need to handle backspace manually
+          start = start - 1;
+        }
```

Comment:
I'd suggest not to mix different implementation in a single code. Lets move them to separate methods, like 

`handleDeleteInNativeInput()`
`handleDeleteInContentEditable()`